specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Pipeline.json

```diff
@@ -1051,6 +1051,29 @@
         }
       }
     },
+    "CosmosDbSqlApiSource": {
```

Comment:
Where are CosmosDbSqlApiSource and CosmosDbSqlApiSink used in your api?

Reply:
It will be used by the "copy" activity and this is the way we extend our support for new data stores in ADF.

Reply:
Hi @shawnxzq, thanks for your explanation. Your PR looks good in general. I have one more question here: Where do you define your copy activity? I cannot find the corresponding operation in your swagger.

Reply:
In pipeline.json, you can check "definitions/CopyActivity", it has properties "source" and "sink" which refer to type "definitions/CopySource" and "definitions/CopySink" respectively. The "CosmosDbSqlApiSource" and "CosmosDbSqlApiSink" is one the "CopySource"/"CopySink" implementations respectively. Basically, this is the way we support various data stores as an data integration service.